test-data/unit/cmdline.pyproject.test

```diff
@@ -0,0 +1,793 @@
+-- Tests for command line parsing
+-- ------------------------------
+--
+-- The initial line specifies the command line, in the format
+--
+--   # cmd: mypy <options>
+--
+-- Note that # flags: --some-flag IS NOT SUPPORTED.
+-- Use # cmd: mypy --some-flag ...
+--
+-- '== Return code: <value>' is added to the output when the process return code
+-- is "nonobvious" -- that is, when it is something other than 0 if there are no
+-- messages and 1 if there are.
+
+-- Directories/packages on the command line
+-- ----------------------------------------
+
+-- Tests mirrored off of cmdline.test
+
+[case testConfigFilePyprojectTOML]
+# cmd: mypy --config-file pyproject.toml main.py
+[file pyproject.toml]
+\[tool.mypy]
+python_version = 2.7
+[file main.py]
+def f():
+    try:
+        1/0
+    except ZeroDivisionError, err:
+        print err
+
+[case testErrorContextConfigPyprojectTOML]
+# cmd: mypy main.py
+[file pyproject.toml]
+\[tool.mypy]
+show_error_context = true
+[file main.py]
+def f() -> None:
+  0 + ""
+[out]
+main.py: note: In function "f":
+main.py:2: error: Unsupported operand types for + ("int" and "str")
+
+[case testNoConfigFilePyprojectTOML]
+# cmd: mypy main.py --config-file=
+[file pyproject.toml]
+\[tool.mypy]
+warn_unused_ignores = true
+[file main.py]
+# type: ignore
+
+[case testPerFileConfigSectionPyprojectTOML]
+# cmd: mypy x.py y.py z.py
+[file pyproject.toml]
+\[tool.mypy]
+disallow_untyped_defs = true
+\[[tool.mypy.overrides]]
+module = "y"
+disallow_untyped_defs = false
+\[[tool.mypy.overrides]]
+module = "z"
+disallow_untyped_calls = true
+[file x.py]
+def f(a):
+    pass
+def g(a: int) -> int:
+    return f(a)
+[file y.py]
+def f(a):
+    pass
+def g(a: int) -> int:
+    return f(a)
+[file z.py]
+def f(a):
+    pass
+def g(a: int) -> int:
+    return f(a)
+[out]
+z.py:1: error: Function is missing a type annotation
+z.py:4: error: Call to untyped function "f" in typed context
+x.py:1: error: Function is missing a type annotation
+
+[case testPerFileConfigSectionMultipleMatchesDisallowedPyprojectTOML]
+# cmd: mypy xx.py xy.py yx.py yy.py
+[file pyproject.toml]
+\[tool.mypy]
+\[[tool.mypy.overrides]]
+module = "*x*"
+disallow_untyped_defs = true
+\[[tool.mypy.overrides]]
+module = "*y*"
+disallow_untyped_calls = true
+[file xx.py]
+def f(a): pass
+def g(a: int) -> int: return f(a)
+[file xy.py]
+def f(a): pass
+def g(a: int) -> int: return f(a)
+[file yx.py]
+def f(a): pass
+def g(a: int) -> int: return f(a)
+[file yy.py]
+def f(a): pass
+def g(a: int) -> int: return f(a)
+[out]
+pyproject.toml: [mypy-*x*]: Patterns must be fully-qualified module names, optionally with '*' in some components (e.g spam.*.eggs.*)
+pyproject.toml: [mypy-*y*]: Patterns must be fully-qualified module names, optionally with '*' in some components (e.g spam.*.eggs.*)
```

Comment:
This error message is a bit confusing (though not a big deal).

Reply:
Agreed. Will fix it up.